pkg/reconciler/v1alpha1/taskrun/resources/taskrunresolution.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either extress or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"fmt"
+
+	"github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1"
+)
+
+// ResolvedTaskRun contains all the data that is needed to execute
+// the TaskRun: the TaskRun, it's Task and the PipelineResources it needs.
+type ResolvedTaskRun struct {
+	Task *v1alpha1.Task
+	// Inputs is a map from the name of the input required by the Task
+	// to the actual Resource to use for it
+	Inputs map[string]*v1alpha1.PipelineResource
+	// Outputs is a map from the name of the output required by the Task
+	// to the actual Resource to use for it
+	Outputs map[string]*v1alpha1.PipelineResource
+}
+
+// GetResource is a function used to retrieve PipelineResources.
+type GetResource func(string) (*v1alpha1.PipelineResource, error)
+
+// GetTask is a function used to retrieve Tasks.
+type GetTask func(string) (*v1alpha1.Task, error)
+
+// ResolveTaskRun looks up CRDs referenced by the TaskRun and returns an instance
+// of TaskRunResource with all of the relevant data populated. If referenced CRDs can't
+// be found, an error is returned.
+func ResolveTaskRun(tr *v1alpha1.TaskRunSpec, gt GetTask, gr GetResource) (*ResolvedTaskRun, error) {
```

Comment:
nit: any reason to use such short variable names ? Using `getTask`, `getResource` would make the following code a bit easier to read :angel:

Reply:
sure I'll change it!

I'm always torn between whether to use shorter or longer names - I feel like Go code in general seems to prefer shorter names, but it looks like I did it at the expense of readability here :D

Reply:
oh nice, that's great advice!! thanks for the link :D